extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/ReadMongoP.java

```diff
@@ -144,7 +144,18 @@ public boolean complete() {
     @Override
     public void close() {
         if (reader != null) {
```

Comment:
Perhaps easier if you let the MongoChunkedReader implement Closable so you can feed it into the IOUtil.closeResource. Then you don't need to do null check or deal with error handling.

Reply:
Good idea, [Implement Closeable](https://github.com/hazelcast/hazelcast/pull/23880/commits/4d6b779db4d7958e3fb31aabef59f687b1c665ec)